### ISSUE TITLE
[3.3.5] DB/Quest: The Exorcism ( Mage Quest 1955)

### ISSUE THREAD

**Issue by Tauriella** [TrinityCore/TrinityCore#15932]
If you accept the quest nothing happens!

TrinityCore rev. 94e4bddb903d 2015-12-01 19:27:17 +0100 (HEAD branch)

http://www.wowhead.com/quest=1955/the-exorcism

**Comment by tkrokli** [comment]
TrinityCore rev. 94e4bdd 2015-12-01 19:27:17 +0100 (3.3.5 branch) (Win64, Release)
TDB 335.60 + updates up to and including 2015_12_01_00_world.sql

Confirmed. Tabetha (NPC ID 6546) does not seem to have a dedicated SAI for quest 1955, only for
http://www.wowhead.com/quest=4961/cleansing-of-the-orb-of-orahil (Warlock quest) and the other Mage quest http://www.wowhead.com/quest=1957/mana-surges. The only thing that happens in this quest (Exorcism, 1955) is that the timer is started when you accept the quest (DB timer value).

I found the following SAI lines while looking for a possible script that could belong to Tabetha:
```sql
SELECT * FROM `smart_scripts` WHERE `entryorguid` LIKE '6546%' ORDER BY `entryorguid`,`id`;
```
```
entryorguid  source_type      id    link  event_type  event_phase_mask  event_chance  event_flags  event_param1  event_param2  event_param3  event_param4  action_type  action_param1  action_param2  action_param3  action_param4  action_param5  action_param6  target_type  target_param1  target_param2  target_param3  target_x  target_y  target_z  target_o  comment                                                                           
-----------  -----------  ------  ------  ----------  ----------------  ------------  -----------  ------------  ------------  ------------  ------------  -----------  -------------  -------------  -------------  -------------  -------------  -------------  -----------  -------------  -------------  -------------  --------  --------  --------  --------  ----------------------------------------------------------------------------------
       6546            0       0       0          19                 0           100            0          4961             0             0             0           80         654600              2              0              0              0              0            1              0              0              0         0         0         0         0  Tabetha - On Quest 'Cleansing of the Orb of Orahil' Taken - Run Script (No Repeat)
       6546            0       1       0          20                 0           100            0          1957             0             0             0            9              0              0              0              0              0              0           14          28294         103680              0         0         0         0         0  Tabetha - On Quest 'Mana Surges' Finished - Activate Gameobject (No Repeat)       
       6546            0       2       3          19                 0           100            0          1957             0             0             0            9              0              0              0              0              0              0           14          28294         103680              0         0         0         0         0  Tabetha - On Quest 'Mana Surges' Taken - Activate Gameobject                      
       6546            0       3       0          61                 0           100            0             0             0             0             0           87         654601         654602         654603              0              0              0            1              0              0              0         0         0         0         0  Tabetha - On Quest 'Mana Surges' Taken - Run Random Script                        
       6546            0       4       0          38                 0           100            0             1             1             0             0           87         654601         654602         654603              0              0              0            1              0              0              0         0         0         0         0  Tabetha - On Data Set 1 1 - Run Random Script                                     
     654600            9       0       0           0                 0           100            0             0             0             0             0            1              0              0              0              0              0              0            1              0              0              0         0         0         0         0  Tabetha - On Script - Say Line 0                                                  
     654600            9       1       0           0                 0           100            0          2000          2000             0             0           11           9097              0              0              0              0              0            1              0              0              0         0         0         0         0  Tabetha - On Script - Cast 'Summon Demon of the Orb'                              
     654601            9       0       0           0                 0           100            0             0             0             0             0           12           6550              1         600000              0              0              0            8              0              0              0  -4019.22  -3383.91   38.2265    2.7963  Tabetha - On Script - Summon Creature 'Mana Surge'                                
     654601            9       1       0           0                 0           100            0             0             0             0             0           12           6550              1         600000              0              0              0            8              0              0              0  -4019.22  -3383.91   38.2265    2.7963  Tabetha - On Script - Summon Creature 'Mana Surge'                                
     654601            9       2       0           0                 0           100            0             0             0             0             0           12           6550              1         600000              0              0              0            8              0              0              0  -4019.22  -3383.91   38.2265    2.7963  Tabetha - On Script - Summon Creature 'Mana Surge'                                
     654602            9       0       0           0                 0           100            0             0             0             0             0           12           6550              1         600000              0              0              0            8              0              0              0  -4019.99  -3394.54   38.5507    1.8342  Tabetha - On Script - Summon Creature 'Mana Surge'                                
     654602            9       1       0           0                 0           100            0             0             0             0             0           12           6550              1         600000              0              0              0            8              0              0              0  -4019.99  -3394.54   38.5507    1.8342  Tabetha - On Script - Summon Creature 'Mana Surge'                                
     654603            9       0       0           0                 0           100            0             0             0             0             0           12           6550              1         600000              0              0              0            8              0              0              0  -4013.29  -3385.14   38.4656    2.3801  Tabetha - On Script - Summon Creature 'Mana Surge'                                
     654603            9       1       0           0                 0           100            0             0             0             0             0           12           6550              1         600000              0              0              0            8              0              0              0  -4013.29  -3385.14   38.4656    2.3801  Tabetha - On Script - Summon Creature 'Mana Surge'                                
     654603            9       2       0           0                 0           100            0             0             0             0             0           12           6550              1         600000              0              0              0            8              0              0              0  -4013.29  -3385.14   38.4656    2.3801  Tabetha - On Script - Summon Creature 'Mana Surge'                                
```
Although when comparing the quest texts from wowhead and looking at both the objective link texts,
- [**Demon of the Orb**](http://www.wowhead.com/npc=6549/demon-of-the-orb) slain
it looks like the exact same (Felguard) demon: http://www.wowhead.com/npc=6549/demon-of-the-orb

Would it be as simple as using the same ideas and links as in the Warlock quest SAI for this Quest too?

**Comment by dr-j** [comment]
Actually may not be right because we lack sniffs for this but there is another quest with exact same requirement (demon of the orb) from same quest giver which is scripted and works.

http://www.wowhead.com/quest=4961/cleansing-of-the-orb-of-orahil  which is warlock class quest but exact same requirement as the mage one

I writ the below quiet some time ago this uses the same script from cleansing the orb of orahill which allows this quest to be completed wether script should be different for this quest is a different story.

~~~code
        -- Fix The Exorcism (Q 1955)
        -- Will get this to run same script as what runs when Cleansing of the Orb of Orahil
        -- Quest giver is same for both quests and the objective is same for both (ie both to kill Demon of the Orb NPC 6549

        DELETE FROM `smart_scripts` WHERE `entryorguid`=6546 AND `id`=5;

        INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
        (6546, 0, 5, 0, 19, 0, 100, 0, 1955, 0, 0, 0, 80, 654600, 2, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Tabetha - On Quest Accept - Run Script');

**Comment by tkrokli** [comment]
Yes, I was about to test something similar, but I fell asleep before I could get it tested and post it here.
I would have added the quest name in the comment field, just to mark it as specific for Quest 1955.

About the wording in the long comment lines; hope it is OK with you if I suggest a modified version:
```sql
-- Fix quest "The Exorcism" (ID 1955) by running the same script as "Cleansing of the Orb of Orahil" (ID 4961).
-- Quest giver and objective are the same for both quests, i.e. both are to kill "Demon of the Orb", NPC 6549

DELETE FROM `smart_scripts` WHERE `entryorguid`=6546 AND `id`=5;
INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
(6546, 0, 5, 0, 19, 0, 100, 0, 1955, 0, 0, 0, 80, 654600, 2, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Tabetha - On Quest \'The Exorcism\' Taken - Run Script (No Repeat)');
```

**Comment by dr-j** [comment]
Lol I writ that text 3 years ago p:  Had to look to find that fix p:

**Comment by tkrokli** [comment]
Heh, OK. I wonder why nobody complained about this quest not working until now, but maybe that is why you have not pushed it or suggested to add it to the source before now.

While looking around for old and existing code, I found the old scripts here in a DB update from @LeGuybrush in [Organize sql/old (2011_11_20_15_world_sai.sql)](https://github.com/TrinityCore/TrinityCore/blob/3bc3b67a8fae059e1019446b09882dfa7fcd11a0/sql/old/3.3.5a/TDB43_to_TDB44_updates/world/2011_11_20_15_world_sai.sql).

BTW, I have tested just now to run the query (works fine) and looked at how it works in-game too. Everything seems to work. I just don't remember if the Felguard is supposed to walk with that strange speed. It looks like how people are walking in the silent movies from the 1920s. Maybe a client thing.
